classes/local/store/s3/client.php

```diff
@@ -486,9 +486,10 @@ public function generate_presigned_url($contenthash, $headers = array()) {
     /**
      * @param string $contenthash
      * @param array $headers
+     * @param bool $nicefilename
      * @return signed_url
      */
-    private function generate_presigned_url_s3($contenthash, $headers) {
+    private function generate_presigned_url_s3($contenthash, array $headers = [], $nicefilename = true) {
```

Comment:
Can you add a unit test for this? It's unclear to see what was wrong with the urls it was creating and how they have changed